### PR TITLE
Check Run Annotations

### PR DESCRIPTION
**Pull request by StanleyGoldman**
This pull request adds functionality to:
- Show a clickable indicator on pull request checks that have annotations
- Show a list of annotations on a pull request
- Show a count of annotation failures, warnings and notices in a pull request file list.

#### Remaining
- [x] Design work

#### Before merging
- [x] merge #1900 into this branch/PR

### Images

image

image

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
@donokuda... send 💕

File: src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+            try
+            {
+                await Task.Delay(0);
+                PullRequestTitle = pullRequest.Title;
+
+                var checkSuiteRun = pullRequest
+                    .CheckSuites.SelectMany(checkSuite => checkSuite.CheckRuns
+                            .Select(checkRun => new{checkSuite, checkRun}))
+                    .First(arg => arg.checkRun.DatabaseId == CheckRunId);
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs]: I don't have a good answer for this yet.

File: src/GitHub.App/SampleData/PullRequestCheckViewModelDesigner.cs [hunk 2]
```diff
@@ -18,5 +19,11 @@ public sealed class PullRequestCheckViewModelDesigner : ViewModelBase, IPullRequ
         public Uri DetailsUrl { get; set; } = new Uri("http://github.com");
 
         public ReactiveCommand<Unit, Unit> OpenDetailsUrl { get; set; } = null;
+
+        public PullRequestCheckType CheckType { get; set; } = PullRequestCheckType.ChecksApi;
```
Comment (StanleyGoldman) [inline src/GitHub.App/SampleData/PullRequestCheckViewModelDesigner.cs]: Add

File: src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (grokys) [reply src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs]: Do you have an answer for this yet?

File: src/GitHub.Exports.Reactive/Models/IInlineAnnotationModel.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿namespace GitHub.Models
+{
+    /// <summary>
+    /// Represents an inline annotation on an <see cref="IPullRequestSessionFile"/>.
+    /// </summary>
+    public interface IInlineAnnotationModel
```
Comment (grokys) [inline src/GitHub.Exports.Reactive/Models/IInlineAnnotationModel.cs]: Is this interface needed? `InlineAnnotationModel` is an immutable class so there's nothing to mock really.

File: src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs]: Fixed.

File: src/GitHub.Exports.Reactive/Models/IInlineAnnotationModel.cs (hunk 3 above)
Comment (StanleyGoldman) [reply src/GitHub.Exports.Reactive/Models/IInlineAnnotationModel.cs]: Fixed

File: src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationItemViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+﻿using System.Linq;
+using System.Reactive;
+using System.Reactive.Linq;
+using GitHub.Models;
+using GitHub.Services;
+using GitHub.ViewModels;
+using GitHub.ViewModels.GitHubPane;
+using ReactiveUI;
+
+namespace GitHub.App.ViewModels.GitHubPane
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationItemViewModel.cs]: No problem, but why aren't we setting the Default Namespace of this project to `GitHub`

File: src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+            IsLoading = true;
+
+            try
+            {
+                LocalRepository = localRepository;
+                RemoteRepositoryOwner = owner;
+                PullRequestNumber = pullRequestNumber;
+                CheckRunId = checkRunId;
+                session = await sessionManager.GetSession(owner, repo, pullRequestNumber);
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply src/GitHub.App/ViewModels/GitHubPane/PullRequestAnnotationsViewModel.cs]: That's no longer the case.

File: src/GitHub.Exports/Models/CheckRunModel.cs [hunk 6]
```diff
@@ -21,7 +28,14 @@ public class CheckRunModel
         /// </summary>
         public DateTimeOffset? CompletedAt { get; set; }
 
-        /// <summary>The name of the check for this check run.</summary>
+        /// <summary>
```
Comment (grokys) [inline src/GitHub.Exports/Models/CheckRunModel.cs]: test

**Comment by donokuda** [comment]
> Is it intended that both comments and annotations now get a grey diamond icon in the margin? I'd have expected the icon to distinguish between comments and annotations

This was intended. I was exploring different ways of indicating if a line had an annotation, a comment, both, and what kind of annotation. Here's a crop of a work-in-progress mockup:

image

The challenge I ran into was how to represent more than one annotation marker in the margin and decided on one to show to simplify things. However, I'll work on iterating in @StanleyGoldman's [pull request](https://github.com/github/VisualStudio/pull/2084) for using different markers.

**Comment by grokys** [comment]
Ok, thanks for the details @donokuda - I've merged this, lets iterate on it in separate PRs!

**Pull request merged**